pandas/tests/frame/methods/test_pct_change.py

```diff
@@ -144,12 +143,15 @@ def test_pct_change_with_duplicated_indices(fill_method):
         {0: [np.nan, 1, 2, 3, 9, 18], 1: [0, 1, np.nan, 3, 9, 18]}, index=["a", "b"] * 3
     )
 
-    msg = (
-        "The 'fill_method' and 'limit' keywords in "
-        "DataFrame.pct_change are deprecated"
-    )
-    with tm.assert_produces_warning(FutureWarning, match=msg):
-        result = data.pct_change(fill_method=fill_method)
+    if fill_method is not None:
+        msg = (
+            "The 'fill_method' keyword being not None and the 'limit' keyword in "
+            "DataFrame.pct_change are deprecated"
+        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            result = data.pct_change(fill_method=fill_method)
+    else:
+        result = data.pct_change(fill_method=None)
```

Comment:
assert_produces_warning (unlike pytest.raises) can accept `None` instead of `FutureWarning`. Can you make this something like:

```
warn = None if fill_method is None else FutureWarning
with tm.assert_produces_warning(warn, match=msg):
   ...
```

Reply:
Amazing feature, never noticed that before.